src/figuresClasses.ts

```diff
@@ -1,11 +1,81 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: 'triangle' | 'circle' | 'rectangle';
+
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
[CODE KNOWLEDGE] - In TypeScript, when you declare a constructor parameter with an access modifier, it implicitly creates a property on the class. You don't need to assign the shape manually in the constructor since 'public shape' is already declared as a parameter.